AsyncAwait.Task1.CancellationTokens/Calculator.cs

```diff
@@ -1,21 +1,27 @@
 ﻿using System.Threading;
+using System.Threading.Tasks;
 
 namespace AsyncAwait.Task1.CancellationTokens;
 
 internal static class Calculator
 {
     // todo: change this method to support cancellation token
-    public static long Calculate(int n /*, CancellationToken token*/)
+    public static async Task<long> CalculateAsync(int n, CancellationToken token)
     {
-        long sum = 0;
-
-        for (var i = 0; i < n; i++)
+        return await Task.Run(() =>
         {
-            // i + 1 is to allow 2147483647 (Max(Int32)) 
-            sum = sum + (i + 1);
-            Thread.Sleep(10);
-        }
-
-        return sum;
+            long sum = 0;
+            for (var i = 0; i < n; i++)
+            {
+                if (token.IsCancellationRequested)
+                {
+                    return -1;
+                }
+                // i + 1 is to allow 2147483647 (Max(Int32)) 
+                sum += (i + 1);
+                Thread.Sleep(10);
```

Comment:
Do not mix non-blocking and blocking code - this is the way toward deadlocks. There is an alternative non-blocking `Task.Delay`

Reply:
fixed

Reply:
@dkepamprotonme `Task.Delay` is awaitable